### PR TITLE
WrappedTypeAnalysisContext.IsValueType

### PR DESCRIPTION
**Pull request by ds5678**

### CODE REVIEW

File: Cpp2IL.Core/Model/Contexts/ArrayTypeAnalysisContext.cs [hunk 1]
```diff
@@ -19,5 +19,7 @@ public ArrayTypeAnalysisContext(Il2CppType rawType, AssemblyAnalysisContext refe
 
     public override string DefaultName => $"{ElementType.Name}[{Rank}]";
 
+    public sealed override bool IsValueType => false;
```
Comment (ds5678) [inline Cpp2IL.Core/Model/Contexts/ArrayTypeAnalysisContext.cs:22]: I made the overrides sealed, but I feel like maybe these classes should be sealed instead. I can't think of a valid reason for someone to derive from them. Thoughts?

File: Cpp2IL.Core/Model/Contexts/PointerTypeAnalysisContext.cs [hunk 2]
```diff
@@ -17,4 +17,6 @@ public PointerTypeAnalysisContext(Il2CppType rawType, AssemblyAnalysisContext re
     public override Il2CppTypeEnum Type => Il2CppTypeEnum.IL2CPP_TYPE_PTR;
 
     public override string DefaultName => $"{ElementType.Name}*";
+
+    public sealed override bool IsValueType => true;
```
Comment (SamboyCoding) [inline Cpp2IL.Core/Model/Contexts/PointerTypeAnalysisContext.cs]: Is this correct?

File: Cpp2IL.Core/Model/Contexts/ArrayTypeAnalysisContext.cs (hunk 1 above)
Comment (SamboyCoding) [reply Cpp2IL.Core/Model/Contexts/ArrayTypeAnalysisContext.cs:22]: Either is probably fine. For now I think leaving the override as sealed will suffice. More flexibility isn't a bad thing.

File: Cpp2IL.Core/Model/Contexts/PointerTypeAnalysisContext.cs (hunk 2 above)
Comment (ds5678) [reply Cpp2IL.Core/Model/Contexts/PointerTypeAnalysisContext.cs]: I thought it was, but apparently not. I'll fix it.

**Pull request merged**